### DISCUSSION TITLE
Runtime Error "ERROR: no system number 0 found!"

### DISCUSSION THREAD

**Discussion by AngeloCass** [idaholab/moose#17350] [Q&A General] [answered]
I have developed a C++ routine that makes use of a MOOSE application to execute a model with varying parameter sets. This routine uses MPI to iterate through different parameter sets in parallel. The routine works as expected on my local machine, but when trying to test on the HPC cluster that I am using, I get the error shown in the subject line. I will attach a text file containing the output with errors.
[Error.txt](https://github.com/idaholab/moose/files/6157853/Error.txt)

**Comment by AngeloCass** [comment]
I forgot to ask this directly, but if anyone has suggestions or solutions for this please let me know! Thanks!

**Comment by loganharbour** [comment]
Can you share the problem (input) you're running?

**Reply by AngeloCass** [reply]
I will attach the input file to this reply. Some of the values in the GlobalParams block are overwritten with command line inputs by the C++ routine at runtime. Also, the Outputs block file base is defined unique to each execution through a command line input. 
[Input.txt](https://github.com/idaholab/moose/files/6158050/Input.txt)

**Comment by aeslaughter** [comment, marked as answer]
Hopefully you are aware that we have native capability for doing this: https://mooseframework.inl.gov/modules/stochastic_tools/index.html
